app/components/AbilitySelect.tsx

```diff
@@ -0,0 +1,43 @@
+import { Select, MenuItem, SelectChangeEvent, Box, Menu } from "@mui/material";
+import Pokedex, { PokemonAbility } from "pokedex-promise-v2";
+import { useEffect, useState } from "react";
+import { toTitleCase } from "@/app/utils/utils";
+
+export function AbilitySelect({
+    pokedex,
+    abilities,
+}: {
+    pokedex: Pokedex;
+    abilities: PokemonAbility[];
+}) {
+    // map names of abilities
+    const abilityNames = abilities.map( i => i.ability.name );
+    // prune dupllicates
+    const filteredAbilities = Array.from( new Set(abilityNames) );
+    const [currentAbilityName, setCurrentAbility] = useState<string>(abilities[0].ability.name);
+
+    function handleChange(event: SelectChangeEvent) {
+        setCurrentAbility(event.target.value);
+    }
+
+    useEffect(() => {
+        pokedex.getAbilityByName(currentAbilityName);
```

Comment:
You don't seem to be doing anything with the return from this call. You can delete it and the useEffect altogether, I think.